src/app/shared-services/db/migrations.ts

```diff
@@ -70,6 +70,10 @@ export const MIGRATIONS = [
     PRIMARY KEY (trajectory, time),
     FOREIGN KEY (trajectory) REFERENCES trajectories(id) ON DELETE CASCADE);`,
 
-  // update date from iso-strings to timestamps
-  `UPDATE points SET time=CAST(strftime('%s', time) AS INT);`,
+  // remove potential duplicate points with very similar timestamp
+  // update date from iso-strings to timestamps and invalidate durationDays for recalculation
+  `DELETE FROM points WHERE time NOT IN (
+    SELECT TIME FROM points GROUP BY strftime('%s', time));
+  UPDATE points SET time=strftime('%s', time);
```

Comment:
In case we still have ISO-strings stored in the database, those are reformatted to seconds since 1970, returned as string and then implicitly converted by some sqlite magic into the datetime-format (which is a number)?

If we already have the correct format (a number representing the seconds since 1970) the aforementioned conversion is still carried out but does effectively not change anything?

Reply:
> In case we still have ISO-strings stored in the database, those are reformatted to seconds since 1970, returned as string and then implicitly converted by some sqlite magic into the datetime-format (which is a number)?

Correct. SQLite is quite limited regrading reformatting stuff, therefore I'm relying on the function `strftime()`. I could additionally reformat/cast it back to a number, but SQLite handles this quite well by itself.

> If we already have the correct format (a number representing the seconds since 1970) the aforementioned conversion is still carried out but does effectively not change anything?

If we already have the correct format (number since 1970), then this migration already happened. This migration does only happen once, therefore this should not be possible.